### PR TITLE
docs: ADR 15 - repo consolidation

### PR DESCRIPTION
**Pull request by bmtcril**
Draft of an ADR to consolidate the Aspects related repositories with a goal of reducing maintenance burden, increasing testability, and simplifying dependencies across the sub-projects.

### CODE REVIEW

File: docs/technical_documentation/decisions/0015_repository_consolidation.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Decisions
+*********
+
+This ADR seeks to address the above issues by consolidating the Aspects project from 5 repositories to 3 by moving the Python code portions into `openedx-aspects`. By making that an installable project which is then used by `tutor-contrib-aspects` we can simplify testing and release of this code, improve code quality, and simplify using Aspects in non-Tutor environments significantly.
+
+**openedx-aspects**
+
+This repo (`openedx-aspects`_) remains the primary Aspects repository where the project decisions and documentation are stored. However, we will also move the Python code for Superset asset management and authentication/authorization into this repository as well as the contents of the `aspects-dbt` and `xapi-db-load` repositories. Those repositories can then be deprecated and archived.
+
+This repo should will also be used to store any common code or configuration between Tutor and non-Tutor Aspects deployments, such as the Alembic migrations needed to bootstrap the Aspects databases.
```
Comment (Ian2012) [inline docs/technical_documentation/decisions/0015_repository_consolidation.rst:30]:
```suggestion
This repo will also be used to store any common code or configuration between Tutor and non-Tutor Aspects deployments, such as the Alembic migrations needed to bootstrap the Aspects databases.
```

File: docs/technical_documentation/decisions/0015_repository_consolidation.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+**tutor-contrib-aspects**
+
+`tutor-contrib-aspects`_ will be updated to use the `openedx-aspects`_ repository as a dependency. This will allow us to keep the Tutor plugin as a separate repository, but will allow us to keep the Python code encapsulated in an easier to test environment, force us to move the Tutor Jinja configuration variables into a traditional configuration layer, where they belong. This will also allow us to greatly simplify the Tutor plugin and separate the concerns of code and configuration.
+
+Care must be taken to ensure that the extension mechanisms for Superset assets, localization, and authentication / authorization are not broken by this change.
```
Comment (Ian2012) [inline docs/technical_documentation/decisions/0015_repository_consolidation.rst:36]: Some (if not most) of those things will be broken, I think we should keep compatibility with custom DBT repositories, but I'm worried that not all extension mechanisms can be kept.

**Review by Ian2012 — APPROVED** [review]
I'm worried that we may not accomplish full compatibility with current extension mechanisms, but it's fine to break it as this will be major refactor and probably we will find better alternatives to extensions

**Comment by bmtcril** [comment]
I think that after some preliminary testing this doesn't end up making sense. The code being developed is run in wildly different contexts (aspects image, superset image, local machine) and putting all of those things and their CI in one place created more confusion for me and more complexity than seems really necessary. There may be a way to consolidate this functionality, but I think this plan confuses concerns too much to pursue right now.

So I'm going to close it and if anyone has other feelings about how we might pursue it in a more sane way we should talk about it!

**Pull request closed without merging**